api

```diff
@@ -351,6 +351,43 @@ app_to_pkgname() { #given an app-name, convert it to a unique, valid package-nam
   echo "pi-apps-$(echo "$app" | md5sum | cut -c1-8 | awk '{print $1}')"
 }
 
+add_external_repo() { # add an external apt repo and its gpg key
+  repopath="$1"
+  reponame="$2"
+  pubkeyurl="$3"
+
+  # check if all needed vars are set
+  [ -z "$repopath" ] && error "add_external_packages: repopath not set"
+  [ -z "$reponame" ] && error "add_external_packages: reponame not set"
+  [ -z "$pubkeyurl" ] && error "add_external_packages: pubkeyurl not set"
+
+  # check if links are valid
+  wget -q --spider "$pubkeyurl" || error "add_external_packages: pubkeyurl isn't a valid link"
+
+  # make pi-apps key storage dir if it doesn't exist
+  if [ ! -d /usr/share/keyrings ]; then
+    mkdir -p /usr/share/keyrings || error "add_external_repo: failed to create pi-apps key storage dir."
```

Comment:
all files/folders in /usr are root owned, this would fail without sudo if it gets called